lib/checknullpointer.cpp

```diff
@@ -509,31 +509,49 @@ void CheckNullPointer::arithmetic()
     for (std::size_t i = 0; i < functions; ++i) {
         const Scope * scope = symbolDatabase->functionScopes[i];
         for (const Token* tok = scope->classStart->next(); tok != scope->classEnd; tok = tok->next()) {
-            if (!tok->astOperand2() || tok->str() != "-")
-                continue;
-            // pointer subtraction
-            if (!tok->valueType() || !tok->valueType()->pointer)
-                continue;
-            // Can LHS be NULL?
-            const ValueFlow::Value *value = tok->astOperand1()->getValue(0);
-            if (!value)
-                continue;
-            if (!_settings->inconclusive && value->isInconclusive())
-                continue;
-            if (value->condition && !_settings->isEnabled(Settings::WARNING))
-                continue;
-            arithmeticError(tok,value);
+            if (tok->str() == "-" || tok->str() == "+" || tok->str() == "+=" || tok->str() == "-=")
+            {
+                const Token *arguments[] = {tok->astOperand1(), tok->astOperand2()};
+                for(const Token *argument:arguments) {
+                    if(argument && argument->valueType() && argument->valueType()->pointer) {
+                        long checkValue = 0;
+                        // When using an assign op, the value read from
+                        // valueflow has already been updated, so instead of
+                        // checking for zero we check that the value is equal
+                        // to RHS
+                        if (tok->astOperand2()->hasKnownIntValue()) {
+                            if (tok->str() == "-=") checkValue -= tok->astOperand2()->values().front().intvalue;
```

Comment:
please write the conditional code on its own line to match the normal cppcheck coding style.

    if (tok->str() == "-=")
        checkValue -= tok->astOperand2()->values().front().intvalue;